File: tabelas-hash/exemplo-tabela-hash.js
const caderno = {} //* implementação da tabela hash em javascript, assim transformamos "caderno" em uma tabela hash

//* Relacionando as strings a números
caderno['maçã'] = 0.67
caderno['leite'] = 1.49
caderno['abacate'] = 1.60

console.log(caderno)
console.log(caderno['leite']) //? executa uma função hash, inserimos uma string e recebemos um número

function acharPreco(item) {
    if(caderno[item]) { //? assim verificamos se o item existe na nossa tabela hash
        console.log(caderno[item]);
    } else {
        throw 'Esse item não existe na tabela' //! caso o item não exista o usuário será informado
    }
}

acharPreco('abacate')


const votaram = {}

const verificarEleitor = nomeEleitor => {
    if(votaram[nomeEleitor]) console.log('Ralá')
    else {
        votaram[nomeEleitor] = true
        console.log('Pode votar')
    }
}

verificarEleitor('joão')
verificarEleitor('joão')

File: tabelas-hash/exemplo-tabela-hash.py
caderno = dict() #implementação da tabela hash em python, assim transformamos "caderno" em uma tabela hash

caderno['maçã'] = 0.67
caderno['leite'] = 1.49
caderno['abacate'] = 1.60

print(caderno)
print(caderno['abacate'])


votaram = {}

def verifica_eleitor(nome_eleitor):
    if votaram.get(nome_eleitor): # se o nome do eleitor já estiver presente na lista das pessoas que votaram ele não pode votar
        print('Ralá')
    else: # se o nome do eleitor não estiver presente na lista ele pode votar
        votaram[nome_eleitor] = True
        print('Pode votar')

verifica_eleitor('joão')
verifica_eleitor('joão')
verifica_eleitor('erlando')